src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -273,7 +299,22 @@ namespace AppInstaller::CLI::Workflow
         {
             const auto& path = context.Get<Execution::Data::InstallerPath>();
             AICLI_LOG(CLI, Info, << "Removing installer: " << path);
-            std::filesystem::remove(path);
+
+            try
+            {
+                // best effort
+                std::filesystem::remove(path);
+            }
+            catch (const std::exception& e)
+            {
+                AICLI_LOG(CLI, Warning, << "Failed to remove installer file after execution. Reason: " << e.what());
+                AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_REMOVE_INSTALLER_FAILED);

```

Comment:
>APPINSTALLER_CLI_ERROR_REMOVE_INSTALLER_FAILED [](start = 40, length = 46)

I'm not sure that this should be an error, at most it seems like it should result in a warning output to the Reporter.

Reply:
I can rename the error code to be APPINSTALLER_CLI_WARNING... and output a warning message. Definitely I don't want E_ACCESSDENIED in the reports to be mixed with others

---
In reply to: [510504068](https://github.com/microsoft/winget-cli/pull/625#discussion_r510504068) [](ancestors = 510504068)

Reply:
I am suggesting that we simply move on without an error of any kind.  We could log to telemetry if we want, but I don't think a failure to clean up fully warrants an error.  Full OS upgrade (not the little CBS ones) will clean files out, and if it does become a problem we can put in some cleanup like I did for logs.

---
In reply to: [510525154](https://github.com/microsoft/winget-cli/pull/625#discussion_r510525154) [](ancestors = 510525154,510504068)

Reply:
Removed the error code and ignored this failure, we can add back reporting if needed in the future

---
In reply to: [511105679](https://github.com/microsoft/winget-cli/pull/625#discussion_r511105679) [](ancestors = 511105679,510525154,510504068)